Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProject/Screens/CourseListVC.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  CourseListViewController.swift
+//  IosInterviewPracticeProject
+//
+//  Created by Robert Ramirez on 10/16/20.
+//  Copyright © 2020 me.robert.ramirez. All rights reserved.
+//
+
+import UIKit
+
+class CourseListVC: UIViewController {
+  private var courseContentUrl: String?
+  private var courses = [Item]()
+  private var dataSource: UITableViewDiffableDataSource<Section, Item>?
+  private let courseManager = CourseAPIService()
+  private let timeFormatter: DateComponentsFormatter = {
+    let timeFormatter = DateComponentsFormatter()
+    timeFormatter.allowedUnits = [.hour, .minute]
+    timeFormatter.unitsStyle = .short
+    return timeFormatter
+  }()
+  @IBOutlet var courseList: UITableView!
+  @IBAction func filterCourses(_ sender: UISegmentedControl) {
+    var filteredContent = [Item]()
+
+    switch Filter(rawValue: sender.selectedSegmentIndex) {
+    case .all:
+      filteredContent = courses
+    case .article:
+      filteredContent = courses.filter { $0.attributes.contentType == ContentType.article }
+    case .collection:
+      filteredContent = courses.filter { $0.attributes.contentType == ContentType.collection }
+    default:
+      filteredContent = courses
+    }
+
+    DispatchQueue.main.async {
+      self.createSnapshot(from: filteredContent.sorted { $0.attributes.releasedAt > $1.attributes.releasedAt })
+    }
+  }
+
+    override func viewDidLoad() {
+      super.viewDidLoad()
+      courseList.delegate = self
+      courseList.dataSource = dataSource
+      configureDataSource()
+      fetchContent()
+    }
+
+  private func configureDataSource() {
+    dataSource = UITableViewDiffableDataSource<Section, Item>(tableView: courseList) { tableView, indexPath, item -> UITableViewCell in
+      guard let cell = tableView.dequeueReusableCell(withIdentifier: CourseListCell.reuseId, for: indexPath) as? CourseListCell else { return UITableViewCell() }
+      let duration = self.timeFormatter.string(from: TimeInterval(Double(item.attributes.duration))) ?? ""
+
+      cell.courseName.text = item.attributes.name
+      cell.courseDescription.text = item.attributes.descriptionPlainText
+      cell.duration.text = "(\(duration))"
+      cell.courseType.text = item.attributes.contentType == ContentType.article ? "Article Course" : "Video Course"
+      cell.releaseDate.text = item.attributes.releasedAt.formatDateString()
+      cell.links = item.links
```

Comment:
I would move all of this setup code into the cell, and expose an `item` property in the cell. You can then set all of the cell's properties by adding a `didSet` to that property. That way the cell is responsible for itself, and the VC is cleaner.